src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java

```diff
@@ -1,28 +1,83 @@
 package ca.bc.gov.open.jag.efilingapi.submission.service;
 
+import ca.bc.gov.ag.csows.accounts.NestedEjbException_Exception;
+import ca.bc.gov.open.jag.efilingaccountclient.EfilingAccountService;
+import ca.bc.gov.open.jag.efilingapi.api.model.GenerateUrlRequest;
+import ca.bc.gov.open.jag.efilingapi.fee.FeeService;
+import ca.bc.gov.open.jag.efilingapi.fee.models.FeeRequest;
+import ca.bc.gov.open.jag.efilingapi.submission.mappers.SubmissionMapper;
 import ca.bc.gov.open.jag.efilingapi.submission.models.Submission;
-import org.springframework.cache.annotation.CachePut;
-import org.springframework.cache.annotation.Cacheable;
+import ca.bc.gov.open.jag.efilingcommons.exceptions.InvalidAccountStateException;
+import ca.bc.gov.open.jag.efilingcommons.exceptions.StoreException;
+import ca.bc.gov.open.jag.efilingcommons.model.AccountDetails;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.cache.CacheProperties;
 
 import java.util.Optional;
 import java.util.UUID;
 
-/**
- * Implementation of SubmissionService using Submission StorageService
- */
 public class SubmissionServiceImpl implements SubmissionService {
 
+    Logger logger = LoggerFactory.getLogger(SubmissionServiceImpl.class);
 
-    @Override
-    @CachePut(cacheNames = "submission", key = "#submission.id", cacheManager = "submissionCacheManager")
-    public Optional<Submission> put(Submission submission) {
-        return Optional.of(submission);
+    private final SubmissionStore submissionStore;
+
+    private final CacheProperties cacheProperties;
+
+    private final SubmissionMapper submissionMapper;
+
+    private final EfilingAccountService efilingAccountService;
+
+    private final FeeService feeService;
+
+    public SubmissionServiceImpl(
+            SubmissionStore submissionStore,
+            CacheProperties cacheProperties,
+            SubmissionMapper submissionMapper,
+            EfilingAccountService efilingAccountService,
+            FeeService feeService) {
+        this.submissionStore = submissionStore;
+        this.cacheProperties = cacheProperties;
+        this.submissionMapper = submissionMapper;
+        this.efilingAccountService = efilingAccountService;
+        this.feeService = feeService;
     }
 
     @Override
-    @Cacheable(cacheNames = "submission", key = "#key", cacheManager = "submissionCacheManager", unless="#result == null")
-    public Optional<Submission> getByKey(UUID key) {
-        return Optional.empty();
+    public Submission generateFromRequest(GenerateUrlRequest generateUrlRequest) throws NestedEjbException_Exception {
+
+        logger.debug("Attempting to get user cso account information");
+        AccountDetails accountDetails = efilingAccountService.getAccountDetails(formatUserGuid(generateUrlRequest.getUserId()));
+        logger.info("Successfully got cso account information");
```

Comment:
should these logs both be the same level?

Reply:
the first one is more to tell you where you are... but we can discuss that, usually debug are not enable so we don't see them...